CMakeLists.txt

```diff
@@ -599,9 +599,38 @@ endif()
 
 # Look for OpenGL here before we potentially switch to looking for static libs.
 if (BUILD_CLIENT)
+    if (LINUX)
+        # Set LEGACY OpenGL ABI if the variable is not explictly set.
+        # The backward-incompatible GLVND OpenGL ABI should only be used
+        # by developers who have a good reason to require it.
+        # See: https://github.com/DaemonEngine/Daemon/issues/945#issuecomment-1835091256
+        set(OpenGL_GL_PREFERENCE "LEGACY" CACHE STRING "Linux OpenGL ABI.")
+    endif()
+
     find_package(OpenGL REQUIRED)
     include_directories(${OPENGL_INCLUDE_DIR})
     set(LIBS_CLIENT ${LIBS_CLIENT} OpenGL::GL)
+
+    if (LINUX)
+        # Set CMake default OpenGL ABI for Linux if the variable is set as empty.
+        # Either this is unsupported by the currently used CMake,
+        # either this is supported but uses default value.
+        if (NOT OpenGL_GL_PREFERENCE)
+            if (NOT OPENGL_gl_LIBRARY)
+                # If the GLVND OpenGL and GLX libraries are available, prefer them.
+                # This forces OPENGL_gl_LIBRARY to be empty.
+                # -- https://cmake.org/cmake/help/latest/module/FindOpenGL.html
+                set(OpenGL_GL_PREFERENCE "GLVND")
+            else()
+                # If OpenGL_GL_PREFERENCE is not supported
+                # then the behaviour is LEGACY one.
+                set(OpenGL_GL_PREFERENCE "LEGACY")
+            endif()
+        endif()
+        message(STATUS "Found OpenGL ABI: ${OpenGL_GL_PREFERENCE}")
```

Comment:
I may use a variable named 'OpenGL_selected_ABI` or something like that instead.

Reply:
I made the code less obvious by naming the variable `OpenGL_selected_ABI`, this variable is meant to reflect the internal state of what CMake will use, either because it is requested by the user, either because CMake picked one or the other.